README.md

```diff
@@ -1,204 +1,270 @@
+We will use [pip](https://pypi.org/project/pip/) (_Package Installer for Python_)
+to install _virtualenv_, in this case, we will use _pip_ in version _19.0.3_.
 
-```shell
-virtualenv ./venv
-source ./venv/bin/activate
+Make sure to [install the pip]((https://pip.pypa.io/en/stable/installing/))
+according to your operating system, then [install virtualenv](https://virtualenv.pypa.io/en/latest/installation.html#via-pip).
+That done, we will be able to create and manage our virtual environment to run
+the service.
+
+First let's create a new _virtual environment_ using the following command:
+
+```bash
+# The most common is to create virtualenv at the
+# root of the project that it will belong to.
+$> virtualenv ./venv
 ```
 
-Which will create a `venv` folder in current directory and install
-basic Python packages there. The second command will set all environment
-variables so that new Python packages are installed there as well.
+After creating a _virtual environment_, we need to activate it so that we can
+install the necessary project packages. For this, we use the following command:
 
-To install all dependencies from Auth, execute the following commands.
+```bash
+$> source ./venv/bin/activate
+```
 
-```shell
-# you may need sudo for those
-apt-get install -y python-pip versiontools
-# There is a package that is not in PyPI
-pip install -r ./requirements/requirements.txt
-python setup.py install
+__NOTE THAT__ to exit the _virtual environment_ we use the `deactivate` command,
+as we can see below:
+
+```bash
+# When you no longer want to run the History service,
+# you need to leave the virtual environment created for it:
+$> deactivate ./venv
 ```
 
-The last command will generate a .egg file and install it into your virtual
-environment.
+To install all dependencies, execute the following commands.
 
-#### Docker Setup
+```bash
+# Some libraries depend on the specific version of the pip:
+$> python -m pip install pip==19.0.3
 
-Another alternative is to use **docker** to run the service. To build the
-container, from the repository's root:
+$> pip install versiontools
 
-```shell
+$> pip install -r ./requirements/requirements.txt
+```
+
+The next and last command will generate a
+[`.egg`](http://peak.telecommunity.com/DevCenter/PythonEggs) file and install
+it into your virtual environment:
+~~~bash
+$> python setup.py install
+~~~
+
+#### **History service on Docker**
+
+Another alternative is to use [Docker](https://www.docker.com/)  to run the
+service. To build the container, from the repository's root:
+
+```bash
 # you may need sudo on your machine:
 # https://docs.docker.com/engine/installation/linux/linux-postinstall/
-docker build -t <tag-for-history-docker> -f docker/history.docker .
+$> docker build -t <tag-for-history-docker> -f docker/history.docker .
 ```
-### How to run the service
 
-To run the standalone history service, just set all needed environment variables and execute:
+### **How to run the History service**
+
```

Comment:
`
Beforehand, you need an already running dojot instance in your machine. Check out the
[dojot documentation](https://dojotdocs.readthedocs.io)
for more information on installation methods.
`